src/streamlink/plugins/zattoo.py

```diff
@@ -130,75 +193,97 @@ def _login(self, email, password, _hello):
             'password': password,
             'remember': 'true'
         }
-
-        res = http.post(login_url, headers=self.headers, data=params, cookies=_hello.cookies)
+        res = http.post(login_url, headers=self.headers, data=params)
         data = http.json(res)
 
         self._authed = data['success']
         if self._authed:
-            self.logger.debug('New Session Data')
-            self._session_attributes.set('beaker.session.id', res.cookies.get('beaker.session.id'), expires=3600 * 24)
-            self._session_attributes.set('pzuid', res.cookies.get('pzuid'), expires=3600 * 24)
-            self._session_attributes.set('power_guide_hash', data['session']['power_guide_hash'], expires=3600 * 24)
+            log.debug('New Session Data')
+            self.save_cookies(default_expires=self.TIME_SESSION)
+            self._session_attributes.set('power_guide_hash',
+                                         data['session']['power_guide_hash'],
+                                         expires=self.TIME_SESSION)
+            self._session_attributes.set(
+                'session_control', True, expires=self.TIME_CONTROL)
             return self._authed
-        else:
-            return None
+        return False
```

Comment:
Couldn't you just always return `self._authed` as it contains the result already? Alternatively you could `raise PluginError("Failed to login, check your username/password")`.